screens/booking/BookingDetail.tsx

```diff
@@ -0,0 +1,284 @@
+      };
+    });
+  };
+
+  const openSetting = () => {
+    setIsSetting(true);
+  };
+  const closeSetting = () => {
+    if (
+      bookingRequest.licensePlate == "" ||
+      !bookingRequest.checkInDate ||
+      !bookingRequest.checkInTime ||
+      !bookingRequest.checkOutTime ||
+      !bookingRequest.checkOutDate
+    ) {
+      Alert.alert("Please fill all required fill");
+    } else {
+      setIsSetting(false);
+    }
+  };
+  useEffect(() => {
+    goToNextPage &&
+      navigation.navigate("BookingSummary", { bookingRequest: bookingRequest });
+    setGoToNextPage(false);
+  }, [goToNextPage]);
+  useEffect(() => {
+    isFocused && setBookingRequest(bookingRequest); //refresh screen
+  }, [isFocused]);
+  const handleNavigation = () => {
+    setGoToNextPage(true);
+  };
+  const handleClickConfirm = () => {
+    if (bookingRequest.slot == "" || bookingRequest.floor == "") {
+      Alert.alert("Please select a slot");
+    } else {
+      handleNavigation();
+    }
+  };
+  const handleClickRecommend = (slot: RecommendedSlotType) => {
+    handleOnChange("slot", slot.slotName);
+    handleOnChange("price", slot.price);
+    handleOnChange("unit", slot.unit);
+    handleNavigation();
+  };
+  const RecommendedSlot: React.FC<RecommendedSlotType> = ({
+    slotName,
+    recommendType,
+    price,
+    unit,
+  }) => {
+    return (
+      <Pressable
+        onPress={handleClickRecommend.bind(this, {
+          slotName,
+          recommendType,
+          price,
+          unit,
+        })}
+      >
+        <View style={styles.recommendSlotContainer}>
+          <View style={styles.rowContainer}>
+            <MaterialCommunityIcons
+              name="alpha-p-circle-outline"
+              style={styles.iconParking}
+              size={20}
+            />
+            <View>
+              <BodyText text={"Slot: " + slotName}></BodyText>
+              <BodyText text={recommendType}></BodyText>
+            </View>
+          </View>
+          <BodyText text={price.toString() + unit}></BodyText>
+        </View>
+      </Pressable>
+    );
+  };
+  return (
+    <BodyContainer innerContainerStyle={styles.screen}>
+      <View style={styles.bookingDetailContainer}>
+        <View style={styles.locationContainer}>
+          <BodyText text={"Engineer building 3, Chulalongkorn"} />
+          <MaterialIcons name="location-pin" size={30} style={styles.pin} />
+        </View>
+        {isSetting ? (
+          <View style={styles.bookingDetailComponentContainer}>
+            <BookingDetailComponent
+              licensePlate={bookingRequest.licensePlate}
+              setLicensePlate={(value: string) => {
+                if (value == "Not found your license plate") {
+                  handleOnChange("licensePlate", "");
+                  navigation.navigate("OtherStack", {
+                    screen: "CarInfoSetup",
+                    params: {
+                      mode: ActionMode.CREATE,
+                    },
```

Comment:
I think you should handle some kind of these stuffs in its component. Note: You can use `useNavigation()` to have a navigation there.